src/DataStructures/DataBox/DataBox.hpp

```diff
@@ -1718,6 +1718,52 @@ SPECTRE_ALWAYS_INLINE constexpr auto mutate_apply(
   mutate_apply(F{}, box, std::forward<Args>(args)...);
 }
 
+// @{
+/*!
+ * \ingroup DataBoxGroup
+ * \brief Perform a mutation to the \ref DataBoxGroup `box`, assigning the
+ * `args` to the tags `MutateTags` in order.
+ */
+template <typename... MutateTags, typename BoxTags, typename... Args>
+SPECTRE_ALWAYS_INLINE constexpr void mutate_assign(
+    // NOLINTNEXTLINE(readability-avoid-const-params-in-decls)
+    const gsl::not_null<DataBox<BoxTags>*> box, Args&&... args) noexcept {
+  static_assert(sizeof...(MutateTags) == sizeof...(args),
+                "The number of arguments passed to `mutate_assign` must be "
+                "equal to the number of tags passed.");
+  const auto assign = [](const auto lhs, auto&& rhs) noexcept {
+    *lhs = std::forward<decltype(rhs)>(rhs);
+  };
+  db::mutate<MutateTags...>(
+      box,
+      [&assign, &args...](const gsl::not_null<
+                          typename MutateTags::type*>... box_args) noexcept {
```

Comment:
Make this `const auto... box_args`, that should be all that's necessary so this function also supports base tags, unless I'm missing something. Please add a test where one one the `MutateTags` is a base tag.